### PR TITLE
OCPBUGS 13867: Adding oc-mirror command and flag reference

### PR DESCRIPTION
**Pull request by bergerhoffer**
Version(s):
4.10 only (4.11+ already have these tables)

Issue:
https://issues.redhat.com/browse/OCPBUGS-13867

Link to docs preview:
https://file.rdu.redhat.com/~ahoffer/2023/OCPBUGS-13867/installing/disconnected_install/installing-mirroring-disconnected.html#oc-mirror-command-reference_installing-mirroring-disconnected

QE review:
- [ ] QE has approved this change.

Additional information:

**The majority of this content is the same as what is already live with newer versions: https://docs.openshift.com/container-platform/4.11/installing/disconnected_install/installing-mirroring-disconnected.html#oc-mirror-command-reference_installing-mirroring-disconnected. Was just updated to add/remove items depending on what was available in 4.10.**

### CODE REVIEW

File: modules/oc-mirror-command-reference.adoc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+|If any non image-pull related error occurs, continue and attempt to mirror as much as possible.
+
+|`--dest-skip-tls`
+|Disable TLS validation for the target registry.
+
+|`--dest-use-http`
+|Use plain HTTP for the target registry.
+
+|`--dry-run`
+|Print actions without mirroring images. This flag is experimental and only works when mirroring to disk.
```
Comment (zhouying7780) [inline modules/oc-mirror-command-reference.adoc:53]: `dry-run` not only works for mirror to disk also work for mirror to mirror .

File: modules/oc-mirror-command-reference.adoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+|`--from <string>`
+|Specify the path to an image set archive that was generated by an execution of oc-mirror to load into a target registry.
+
+|`-h`, `--help`
+|Show the help.
+
+|`--log-level <string>`
+|Specify the log level, such as `debug`, `info`, `warn`, or `error`. The default value is `info`.
```
Comment (zhouying7780) [inline modules/oc-mirror-command-reference.adoc:62]: Number for the log level verbosity (valid 1-9, default is 0)

File: modules/oc-mirror-command-reference.adoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+|`--manifests-only`
+|Generate manifests for `ImageContentSourcePolicy` objects to configure a cluster to use the mirror registry, but do not actually mirror any images.
```
Comment (zhouying7780) [inline modules/oc-mirror-command-reference.adoc:65]: `--manifests-only`  only works for an archive with --from, better to highlight it .

Comment (bergerhoffer) [reply modules/oc-mirror-command-reference.adoc:65]: I will create a separate PR to update this for all versions (since this PR is only for 4.10)

Comment (bergerhoffer) [reply modules/oc-mirror-command-reference.adoc:65]: Created #60888 for this followup item

File: modules/oc-mirror-command-reference.adoc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+|`--skip-verification`
+|Skip digest verification.
+
+|`--source-skip-tls`
+|Disable TLS validation for the source registry.
+
+|`--source-use-http`
+|Use plain HTTP for the source registry.
+
```
Comment (dfitzmau) [inline modules/oc-mirror-command-reference.adoc:84]: Not sure if this is intentional, but the following items are missing from 4.10 when compared to 4.11:

Screenshot from 2023-06-07 11-58-31

Comment (dfitzmau) [reply modules/oc-mirror-command-reference.adoc:84]: The 4.10 docs do link to https://docs.openshift.com/container-platform/4.10/updating/updating-restricted-network-cluster/index.html

File: modules/oc-mirror-command-reference.adoc [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+|`completion`
+|Generate the autocompletion script for the specified shell.
+
+|`describe`
+|Output the contents of an image set.
+
+|`help`
+|Show help about any subcommand.
+
```
Comment (dfitzmau) [inline modules/oc-mirror-command-reference.adoc:25]: The `init` parameter and description is missing when compared to the same table as 4.11 docs. Maybe this is intentional?



init | Output an initial image set configuration template.
-- | --





init | Output an initial image set configuration template.
-- | --

**Comment by dfitzmau** [comment]
H @bergerhoffer . Thanks for raising the follow-up PR. I added some comments inline. Should the QE review box be ticked?

File: modules/oc-mirror-command-reference.adoc (hunk 5 above)
Comment (bergerhoffer) [reply modules/oc-mirror-command-reference.adoc:25]: Yes that's correct. --init was introduced in 4.11.

**Comment by bergerhoffer** [comment]
Thanks for reviewing @dfitzmau! No, QE has not reviewed/approved this PR yet, so the QE checkbox shouldn't be checked yet. Indicating QE approval will be more important when it's time for merge review.

File: modules/oc-mirror-command-reference.adoc [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+|`list`
+|List available platform and Operator content and their version.
+
+|`version`
+|Output the oc-mirror version.
```
Comment (mburke5678) [inline modules/oc-mirror-command-reference.adoc:30]: Does oc-mirror here need markup?

File: modules/oc-mirror-command-reference.adoc [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+|Disable TLS validation for the target registry.
+
+|`--dest-use-http`
+|Use plain HTTP for the target registry.
+
+|`--dry-run`
+|Print actions without mirroring images.
+
+|`--from <string>`
+|Specify the path to an image set archive that was generated by an execution of oc-mirror to load into a target registry.
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply modules/oc-mirror-command-reference.adoc:56]: This sentience is a touch convoluted. Woudl two sentences read better?  

Specify the path to an image set archive to load into a target registry. The archive must have been generated by an execution of oc-mirror .

**Comment by mburke5678** [comment]
@bergerhoffer A couple of nits. Otherwise LGTM

FRY @dfitzmau

**Pull request merged**